website/docs/registry/faq.mdx

```diff
@@ -0,0 +1,49 @@
+
+There may be situations where it is important that a version be removed from the Registry. Perhaps a version of a provider has a critical security vulnerability, or you may have extremely high confidence that no one is using a version of your module that doesn't work as you intended it to.
+
+In these situations, your options for removing previously published versions vary depending on the type of integration:
+
+- **Modules**: The owner of a module can use the Terraform Registry website to remove a specific version, all versions of a specific module provider, or remove the module entirely. [More details on this workflow can be found here](/registry/modules/publish#removing-published-content). or [here for policy libraries].
+- **Providers**: There is no manual process for removing a version. Reach out to us via email if it is critical that a version be deleted.
+- **Policy Libraries**: The owner of a policy library can completely remove it from the Registry, through the website. [More details on this workflow can be found here](/registry/policy-libraries/publishing#removing-a-policy-library).
+
+### Can I remove a version of a provider by deleting the release from GitHub?
+
+**We recommend you never remove content from a GitHub repository that has already been published to the Terraform Registry**.
+
+The Terraform Registry acts as a kind of index for Terraform-related content that is hosted on GitHub. When you make a new tag or a new release in a GitHub repository that holds your Terraform integration, the Registry automatically creates an index record for a new content. Think of that index record as simply a pointer to the files that exist in your GitHub repository.
+
+If you remove a file, or a tag, a release, or other content from GitHub, the Registry will not be made aware that you did so. The Registry may still host index records pointing to the now-removed files or content. As a result, someone using your provider or module may see their Terraform configuration break, since through the Registry it expects to find files on GitHub which are no longer available.
+
+### What does it mean to "resync" my provider, module, or policy library?
+
+Owners of any of these types of Terraform integrations can find a "resync" button on the webpage for the integrations they own. Triggering a resync instructs the Registry to analyze the integration's source repository on GitHub. During a resync, the registry will:
+
+- Ensure that there is a webhook established on the repository, so that new versions of the integration can be automatically published.
+- Create any versions of the integration that exist in the GitHub repository but do not yet exist in the Registry.
+
+Please note that the resync procedure will not remove or update versions of the integration that have already been published.
```

Comment:
I dont think this holds for modules. Module versions are mutable, and upserted on every resync. See https://github.com/hashicorp/terraform-registry/blob/947f8e9a8b85cd09a27715e8db5cafb879e2fa88/ingress/github/resync.go#L50-L78

Reply:
Thanks for pointing this out. The divergent behavior between modules and providers is confusing, and it leads to a problem when in putting together docs like this.

We would probably all agree that the behavior of resync should be predictable across all kinds of registry integrations. So should we:

- Document the divergent behavior first, then change that behavior and the docs later so that it is predictable?
- Document the divergent behavior and then never change it, so that we don't confuse users later?
- Make the behaviors identical first and then write the docs for the predictable behavior that we want to have?

Reply:
I suggest - document the divergent behaviour, indicate that work is planned to have it converge (in the docs), then update it when the behaviour has converged.